app/src/main/java/com.example.weatherapp/mvp/presenter/WeatherPresenter.kt

```diff
@@ -9,13 +10,21 @@ class WeatherPresenter(
     private val view: WeatherContract.View
 ) : WeatherContract.Presenter {
 
-    override fun getForecast() {
+    override fun getForecast(onForecastListener: WeatherDayListener) {
```

Comment:
it not necessary that you pass this listener as a parameter, you can access it from the view directly

Reply:
Done